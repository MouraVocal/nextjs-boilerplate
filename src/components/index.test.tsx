import '@testing-library/jest-dom'
import 'jest-styled-components'
import { render, screen } from '@testing-library/react'
import { Main } from '.'

describe('Main', () => {
  it('renders a heading', () => {
    render(<Main />)

    const main = screen.queryByRole('main')

    expect(main).toBeInTheDocument()
  })

  it('should have text', () => {
    render(<Main />)

    const main = screen.queryByRole('main')

    expect(main).toHaveTextContent('Main')
  })

  it('should match snapshot', () => {
    render(<Main />)

    const container = screen.queryByRole('main')

    expect(container).toMatchSnapshot()
    expect(container).toHaveStyle('background: #121212')
  })
})
