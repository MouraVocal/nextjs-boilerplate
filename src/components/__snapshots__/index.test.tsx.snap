// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
.c0 {
  background: #121212;
  color: #e0e0e0;
}

<main
  class="c0"
>
  Main
</main>
`;
